includes/plugins/class-newspack-newsletters.php

```diff
@@ -195,8 +195,22 @@ function( $acc, $item ) {
 	}
 
 	/**
-	 * Ensure the contact is always added to ActiveCampaign's selected master
-	 * list.
+	 * Get lists without the master list, if set.
+	 *
+	 * @param int[] $list_ids List IDs to filter.
+	 */
+	public static function get_lists_without_active_campaign_master_list( $list_ids ) {
+		$master_list_id = Reader_Activation::get_setting( 'active_campaign_master_list' );
+		if ( is_int( intval( $master_list_id ) ) ) {
+			if ( ( $key = array_search( $master_list_id, $list_ids ) ) !== false ) { // phpcs:ignore Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure
```

Comment:
Nit: it took me minute to understand this line—why not use `array_values( array_filter() )` here? Also, I think the wrapping `array_values` might be needed in case the master list isn't at the end of the array—I'm not sure `unset` will re-index the other numeric keys.

Reply:
> why not use array_values( array_filter() ) here? 

Hm, I think both solutions are suboptimal in terms of readability – with `array_filter` after finding the element, you still need the key to unset it 🤷. So we can stick with this one.

Added key reindexing and some tests in 828b26d06e2a514e1bd6da58a177ddd7c0eca8a6

Reply:
Your solution works great with the addition of `array_values`. 👍 

With `array_filter` you wouldn't need to `unset` at all:

```php
$list_ids = array_values(
	array_filter(
		$list_ids,
		function( $list_id ) use ( $master_list_id ) {
			return $list_id !== $master_list_id;
		}
	)
);
```

Reply:
Oh, I see now. Refactored in 91a5122e61758694b5ff3d7a86fa45f94045705e. Thanks!